### PR TITLE
docs: add per-publisher content tabs

### PR DESCRIPTION
**Pull request by woodruffw**
Per https://github.com/pypi/warehouse/pull/15144#issuecomment-1879450151: this adds content tabs (currently just a "GitHub Actions" tab) in a few places where the Trusted Publisher documentation will vary by type.

### CODE REVIEW

File: docs/user/trusted-publishers/creating-a-project-through-oidc.md
Comment (facutuesca) [inline docs/user/trusted-publishers/creating-a-project-through-oidc.md:1]: Nitpick, and unrelated to this PR, but the sentence
`The process for configuring a "pending" publisher are similar to those for
a normal publisher,`
could be changed to
`The steps for configuring a "pending" publisher are similar to those for
a normal publisher,`

File: docs/user/trusted-publishers/using-a-publisher.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Publishing with a Trusted Publisher
 
 # Publishing with a Trusted Publisher
 
-## The easy way
+## GitHub Actions: the easy way
```
Comment (facutuesca) [inline docs/user/trusted-publishers/using-a-publisher.md:7]: Should we have tabs for this page too?

File: docs/user/trusted-publishers/creating-a-project-through-oidc.md
Comment (woodruffw) [reply docs/user/trusted-publishers/creating-a-project-through-oidc.md:1]: That's clearer, thanks! Changing.

File: docs/user/trusted-publishers/using-a-publisher.md (hunk 1 above)
Comment (di) [reply docs/user/trusted-publishers/using-a-publisher.md:7]: Agreed

**Comment by di** [comment]
I think we also need to do something for the "Security Model and Considerations" page, which is very GitHub-focused right now.

**Comment by woodruffw** [comment]
> I think we also need to do something for the "Security Model and Considerations" page, which is very GitHub-focused right now.

Sounds good, I'll make those changes this afternoon.

File: docs/user/trusted-publishers/security-model.md [hunk 2]
```diff
@@ -41,80 +17,119 @@ is not a panacea. In particular:
...
+  be controllable by untrusted parties.
+
+In summary: treat your trusted publishers *as if* they were API tokens. If you
+wouldn't let a user or piece of code access your API token, then they shouldn't
+be able to invoke your trusted publisher.
+
+## Provider-specific considerations
+
+Each trusted publishing provider is its own OIDC identity provider, with its
+own configuration requirements.
```
Comment (di) [inline docs/user/trusted-publishers/security-model.md:30]:
```suggestion
own security model and considerations.
```

**Review by di — APPROVED** [review]

**Pull request merged**